source/docs/software/telemetry/persistent-alerts.rst

```diff
@@ -0,0 +1,55 @@
+# Recording Faults with Persistent Alerts
+
+Robots encounter a variety of fault conditions: disconnected sensors/motors, invalid mechanism states, initialization failures, etc. While the FRC Driver Station provides a :ref:`console interface <docs/software/driverstation/driver-station:Messages Tab>` for instantaneous alerts, the risk of missing important messages makes it poorly suited to communicate faults that persistent over time.
+
+Instead, WPILib provides an API for managing persistent alerts published via Network Tables. Alerts are assigned a priority (*error*, *warning*, or *info*) and can be *activated* or *deactivated* in robot code. The set of active alerts can be displayed on a dashboard, as shown below in Shuffleboard.
+
+.. image:: images/alerts.png
+   :alt: A screenshot of the alerts widget in Shuffleboard, with several active alerts.
+
+Active alerts are automatically displayed in order based on priority and how recently they were activated, with newer and more critical alerts at the top of the list. This provides a mechanism for drivers (on the field) and programmers (off the field) to quickly assess the state of the robot and determine whether it is in a "match-ready" state. Highly descriptive alerts also allow drive teams to adapt to faults in real time, such as switching strategies in case of a subsystem failure.
+
+## API Usage
+
+Alerts are created and managed using the ``Alert`` class. Typically, an alert should be instantiated once and stored as a class member. Periodically, the ``set`` method should be called to update the state of the alert based on relevant data.
```

Comment:
What is the thinking behind this periodic suggestion?  What things do we expect teams to update?

Seems to me most alerts would be activate when it starts and deactivate when it stops and I don't see the use case for the periodic updates in between.  Is there some sort of timeout or something on the alert that I missed?  Wouldn't this just create a swapping effect of the alerts in the widget not really related to when they actually started.

Reply:
I guess my suggestion here might be to phrase this in a way that encourages the user to periodically use the set for simplicity sake.